File: source/java/com/glympse/java/hal/GBatteryProvider.java
//------------------------------------------------------------------------------
//
// Copyright (c) 2014 Glympse Inc.  All rights reserved.
//
//------------------------------------------------------------------------------

package com.glympse.java.hal;

import java.util.*;
import com.glympse.java.core.*;
import com.glympse.java.hal.*;
    
/*J*/public/**/ interface GBatteryProvider extends GCommon
{    
    public void setBatteryListener(GBatteryListener listener);
    
    public void start();
    
    public void stop();    
    
    public int getLevel();
    
    public boolean isPlugged();    
    
    public boolean isPresent();
    
    public void acquireWakeLock();
    
    public void releaseWakeLock();
};  
    
/*C*typedef GBatteryProvider  GBatteryProvider;**/
    

File: source/csharp/glympse/hal/GBatteryProvider.cs
//------------------------------------------------------------------------------
//
// Copyright (c) 2014 Glympse Inc.  All rights reserved.
//
//------------------------------------------------------------------------------

using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;
using System.Threading;

namespace Glympse 
{
    
/*J*public**/ interface GBatteryProvider : GCommon
{    
     void setBatteryListener(GBatteryListener listener);
    
     void start();
    
     void stop();    
    
     int getLevel();
    
     bool isPlugged();    
    
     bool isPresent();
    
     void acquireWakeLock();
    
     void releaseWakeLock();
};  
    
/*C*typedef GBatteryProvider  GBatteryProvider;**/
    
}

